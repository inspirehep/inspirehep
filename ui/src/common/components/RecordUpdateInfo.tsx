import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment-timezone';
import { LOCAL_TIMEZONE } from '../constants';

const DATE_AND_TIME_DISPLAY_FORMAT = `MMM D, YYYY`;

function RecordUpdateInfo({ updateDate }) {
  const formattedDate = moment
    .utc(updateDate)
    .tz(LOCAL_TIMEZONE)
    .format(DATE_AND_TIME_DISPLAY_FORMAT);
  return <span className="light-silver">Updated on {formattedDate}</span>;
}

RecordUpdateInfo.propTypes = { updateDate: PropTypes.string.isRequired };

export default RecordUpdateInfo;
