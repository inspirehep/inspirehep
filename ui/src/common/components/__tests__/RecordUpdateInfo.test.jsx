import React from 'react';
import { shallow } from 'enzyme';

import RecordUpdateInfo from '../RecordUpdateInfo';

describe('RecordUpdateInfo', () => {
  it('renders RecordUpdateInfo with correct update time', () => {
    const testDate = '2020-10-05T13:39:19.083762+00:00';
    const wrapper = shallow(<RecordUpdateInfo updateDate={testDate} />);
    expect(wrapper).toMatchSnapshot();
  });
});
