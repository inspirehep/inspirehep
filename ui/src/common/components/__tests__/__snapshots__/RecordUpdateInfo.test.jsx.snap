// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecordUpdateInfo renders RecordUpdateInfo with correct update time 1`] = `
<span
  className="light-silver"
>
  Updated on
  Oct 5, 2020
</span>
`;
